src/lib/operators/projection.cpp

```diff
@@ -290,4 +295,45 @@ std::shared_ptr<Table> Projection::dummy_table() {
   return shared_dummy;
 }
 
+/**
+ *  Method to determine PQPColumns to forward. In most cases, forwarding a column comes with almost zero costs.
+ *  However, in cases where the same PQPColumn is also accessed in the expression evaluation, the whole column is
```

Comment:
"However"? That is still the case wrt the aggregate operator. It's cleared up in the following sentence, but maybe we can make this one clearer, too.

Reply:
```
/**
 *  Method to determine PQPColumns to forward. As explained above, we forward columns which are simply projected and
 *  which must not be evaluated. But there are cases when forwarding is not beneficial. When a forwardable column is
 *  also evaluated in an expression, the expression evaluator materializes this column and caches it. In case of having
 *  a reference segment as input, forwarding the materialized and cached segment has a similar performance in the
 *  projection operator, but is faster in the following operator. The reason is that the following operator does not
 *  need to process the forwarded reference segment via its position list indirection but can directly access the value
 *  segment sequentially.
 */
ExpressionUnorderedSet Projection::_determine_forwarded_columns(const TableType table_type) const { ...
```

Reply:
> we forward columns THAT are simply projected and NEED not be evaluated.